tests/components/modbus/test_modbus_switch.py

```diff
@@ -184,42 +172,97 @@ async def test_register_switch(hass, regs, expected):
     assert state == expected
 
 
-@pytest.mark.parametrize(
-    "regs,expected",
-    [
-        (
-            [0x40],
-            STATE_ON,
-        ),
-        (
-            [0x04],
-            STATE_OFF,
-        ),
-    ],
-)
-async def test_register_state_switch(hass, regs, expected):
-    """Run test for given config."""
-    switch_name = "modbus_test_switch"
-    state = await base_test(
-        hass,
-        {
-            CONF_NAME: switch_name,
-            CONF_ADDRESS: 1234,
-            CONF_SLAVE: 1,
-            CONF_COMMAND_OFF: 0x04,
-            CONF_COMMAND_ON: 0x40,
-            CONF_VERIFY: {},
+async def test_restore_state_switch(hass):
+    """Run test for sensor restore state."""
+
+    switch_name = "test_switch"
+    entity_id = f"{SWITCH_DOMAIN}.{switch_name}"
+    test_value = STATE_ON
+    config_switch = {CONF_NAME: switch_name, CONF_ADDRESS: 17}
+    with mock.patch(
+        "homeassistant.components.modbus.switch.ModbusSwitch.async_get_last_state"
```

Comment:
Should we use the restore state cache test helper?

Reply:
Yes ! you actually beat me (the dog was walking me :grin:), I will mark this PR as draft, because I have seen some problems in the test harness due to the new async, that I want to solve properly.

Thanks for taking a look.

Reply:
This PR is back for review again, now it works like a charm. Looking forward to hear if you have more review comments.